app/models/concerns/pageflow/uploaded_file.rb

```diff
@@ -55,5 +55,12 @@ def cache_key
       # prevent caching outdated information.
       "#{super}-#{state}"
     end
+
+    def direct_upload_config
+      # the direct uploaad config is only needed for the initial upload.
+      # It can be omitted once the instance leaves the initial "uploading" state
+      return {} unless uploading?
+      Pageflow.config.paperclip_direct_upload_options.call(attachment)
+    end
```

Comment:
Thinking some more about this, I'm not convinced this is the best place to handle this case distinction. The method name suggests that this always returns an upload config. The fact that the comment is needed is another indicator.

The important questions is not about how we create an upload config, but about when we actually present it to the client. So I think the `pageflow/editor/files/_file.json.jbuilder` template would be a much better place for that. This would also give us a decent place to test this logic in a [controller spec](https://github.com/codevise/pageflow/pull/1076/files#diff-547d0353f9388654f30f0ba308e0d3f3R199)

Reply:
changed and covered with spec in f0770050193e889ec9c9a5adb61e8094cad926f4